android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt

```diff
@@ -24,41 +26,63 @@ class CreateListingPageViewModel @Inject constructor(
 ) : SubletrViewModel<CreateListingPageUiState>() {
 	val navHostController: NavHostController get() = navigationService.getNavHostController()
 
-	val addressLineStream: BehaviorSubject<String> = BehaviorSubject.createDefault("")
-	val addressCityStream: BehaviorSubject<String> = BehaviorSubject.createDefault("")
-	val addressPostalCodeStream: BehaviorSubject<String> = BehaviorSubject.createDefault("")
+	val fullAddressStream: BehaviorSubject<String> = BehaviorSubject.createDefault("")
+
+	private val addressStream: Observable<CreateListingPageUiState.AddressModel> = fullAddressStream
+		.observeOn(Schedulers.computation())
+		.map {
+			val splitAddress = it.split(",").toTypedArray()
+
+			CreateListingPageUiState.AddressModel(
+				fullAddress = it,
+				addressLine = if (splitAddress.isNotEmpty()) splitAddress[0] else "",
+				addressCity = if (splitAddress.size >= 2) splitAddress[1] else "",
+				addressPostalCode = if (splitAddress.size >= 3) splitAddress[2] else "",
+				addressCountry = if (splitAddress.size >= 4) splitAddress[3] else "Canada",
+			)
+		}
+		.observeOn(Schedulers.io())
+		.onErrorResumeWith(Observable.never())
 
 	val descriptionStream: BehaviorSubject<String> = BehaviorSubject.createDefault("")
 	val priceStream: BehaviorSubject<Int> = BehaviorSubject.createDefault(0)
 	val numBedroomsStream: BehaviorSubject<Int> = BehaviorSubject.createDefault(0)
 	val startDateStream: BehaviorSubject<String> = BehaviorSubject.createDefault("")
 	val endDateStream: BehaviorSubject<String> = BehaviorSubject.createDefault("")
 
-	val imagesByteStream: BehaviorSubject<MutableList<String>> = BehaviorSubject.createDefault(ArrayList())
+	val imagesBitmapStream: BehaviorSubject<MutableList<Bitmap?>> = BehaviorSubject.createDefault(ArrayList())
```

Comment:
Oh, just a small thing, can you use `mutableListOf()` instead?
It's the more Kotlin-y way of doing things and Kotlin could technically change mutable lists in the future to not use ArrayLists.